cmd/internal/datastore/s3/transaction.go

```diff
@@ -0,0 +1,53 @@
+package s3
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/mailchain/mailchain/stores/s3store"
+)
+
+type TransactionStore struct {
+	*s3store.S3Store
+}
+
+// NewS3TransactionStore creates a new S3 store.
+func NewS3TransactionStore(region, bucket, id, secret string) (*TransactionStore, error) {
+	s3Store, err := s3store.NewS3Store(region, bucket, id, secret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TransactionStore{S3Store: s3Store}, nil
+}
+
+// Key value of resource stored.
+func (sts *TransactionStore) Key(hash []byte) string {
+	return sts.EncodeKey(hash)
+}
+
+type rawTransactionData struct {
+	Protocol string      `json:"protocol"`
+	Network  string      `json:"network"`
+	Hash     []byte      `json:"hash"`
+	Tx       interface{} `json:"transaction"`
+}
+
+func (sts *TransactionStore) PutRawTransaction(ctx context.Context, protocol, network string, hash []byte, tx interface{}) error {
+	rtd := rawTransactionData{
+		Protocol: protocol,
+		Network:  network,
+		Hash:     hash,
```

Comment:
The hash should be encoded otherwise it becomes difficult to read. I would suggest `encoding.EncodeZeroX`